### PR TITLE
Multiply with element of commutative ring

### PR DESCRIPTION
**Pull request by mohamed-barakat**
I need this in `FunctorCategories`.

### CODE REVIEW

File: CAP/gap/ConstructiveCategoriesRecord.gi [hunk 1]
```diff
@@ -18,7 +18,9 @@ IsEnrichedOverCommutativeRegularSemigroup := Concatenation(
 IsAbCategory := Concatenation( [ 
   "ZeroMorphism", 
   "IsZeroForMorphisms", 
-  "AdditiveInverseForMorphisms" ], ~.IsEnrichedOverCommutativeRegularSemigroup ),
+  "SubtractionForMorphisms",
+  "AdditiveInverseForMorphisms",
+  "MultiplyWithElementOfCommutativeRingForMorphisms" ], ~.IsEnrichedOverCommutativeRegularSemigroup ),
```
Comment (sebasguts) [inline CAP/gap/ConstructiveCategoriesRecord.gi]: @sebastianpos Is this correct?

**Comment by sebasguts** [comment]
I do not understand how this command is
* general enough to be in the CAP-kernel
* well-defined from it's documentation/definition

How I see it, if it should be part of the CAP-kernel, there should be more explanation:
* Which commutative ring?
* What is it supposed to do with the morphism?
* Should I implement partial functions for several commutative rings?

I think this should be part of an additional package. Furthermore, I do not think `IsAbCategory` should depend on it.

**Comment by mohamed-barakat** [comment]
What I need in the package `Bialgebroids` is the support for R-linear categories (where R is any constructive commutative ring R): The Hom-set is an R-module and the composition is R-bilinear. I added it to `~.IsEnrichedOverCommutativeRegularSemigroup` as the action of R can be generalized to  commutative regular semigroups.

Another example (still R commutative): A = LeftPresentations( R ) and its generalized morphisms category can then be endowed with this extra R-linear structure. This allows multiplying morphisms in A or G(A) with ring elements in R.

File: CAP/gap/ConstructiveCategoriesRecord.gi (hunk 1 above)
Comment (sebastianpos) [inline CAP/gap/ConstructiveCategoriesRecord.gi]: This list should contain prerequisites for being an Ab-category.
So "MultiplyWithElementOfCommutativeRingForMorphisms"
should not be in this list.

**Comment by sebastianpos** [comment]
Multiplying generalized morphisms with scalars sounds interesting.
I assume you have in mind that you want to multiply the associated morphism
with the given scalar and leave the domain and codomain as they are?

This would turn the Hom-sets of G(A) into something like "R-modules having an underlying commutative regular semigroup instead of an abelian group". Do you know if there is a theory describing such a construction? If there is, then this would give the appropriate context
for your request.

File: CAP/gap/ConstructiveCategoriesRecord.gi (hunk 1 above)
Comment (mohamed-barakat) [inline CAP/gap/ConstructiveCategoriesRecord.gi]: I added `"MultiplyWithElementOfCommutativeRingForMorphisms"` to this list, as I found `"AdditiveInverseForMorphisms"` at the end of it, and commutative regular semigroups do not have additive inverses in the strict sense. Maybe I should create a new list with a name like `IsEnrichedOverRlinearCommutativeRegularSemigroup`.

**Comment by mohamed-barakat** [comment]
Yes, this is what I have in mind.

I interpret the entry `"AdditiveInverseForMorphisms"` in the list `IsEnrichedOverCommutativeRegularSemigroup` as a first step in this direction, as it enables an action of the integers.

And regarding you question on a theory describing such a construction:

In our paper on Gabriel morphisms (https://arxiv.org/pdf/1409.2028.pdf) we write below Lemma A.2:

Hence, a commutative inverse semigroup H defines a direct system of Abelian groups ... over the meet-lattice E(H). In particular ...

On can either define what an R-linear commutative regular/inverse semigroup means (which is straightforward), or replace "Abelian groups" by "R-modules" in the above description.

File: CAP/gap/ConstructiveCategoriesRecord.gi (hunk 1 above)
Comment (sebastianpos) [inline CAP/gap/ConstructiveCategoriesRecord.gi]: Again, this is the list for Ab-categories.
Thus, `"AdditiveInverseForMorphisms"` is correctly inserted here.

**Comment by sebastianpos** [comment]
It would be nice to have a name which is already used
somewhere in the literature.

File: CAP/gap/ConstructiveCategoriesRecord.gi (hunk 1 above)
Comment (mohamed-barakat) [inline CAP/gap/ConstructiveCategoriesRecord.gi]: Yes, sorry, I misinterpreted the piece of code I see above. This list `IsEnrichedOverCommutativeRegularSemigroup` is appended at the end.

**Comment by mohamed-barakat** [comment]
I searched but was unable to find one.

**Pull request closed without merging**